status/status.go

```diff
@@ -77,29 +77,29 @@ type ErrorCountMapMap map[string]ErrorCountMap
 type Status struct {
 	// ImageCount is a map counting the number of images in use indexed by
 	// the image name
-	ImageCount MapMapCount
+	ImageCount MapMapCount `json:"image_count"`
 
 	// PhaseCount is a map counting the number of pods in each phase
 	// (running, failing, scheduled, ...)
-	PhaseCount MapMapCount
+	PhaseCount MapMapCount `json:"phase_count"`
 
 	// PodState counts the number of pods matching conditions such as
 	// desired, ready, available, and unavailable
-	PodState PodStateMap
+	PodState PodStateMap `json:"pod_state"`
 
 	// PodsCount is the number of pods in the k8s cluster
 	// all pods, and pods managed by cilium
-	PodsCount PodsCount
+	PodsCount PodsCount `json:"pods_count"`
 
-	CiliumStatus CiliumStatusMap
+	CiliumStatus CiliumStatusMap `json:"cilium_status"`
 
 	// Errors is the aggregated errors and warnings of all pods of a
 	// particular deployment type
-	Errors ErrorCountMapMap
+	Errors ErrorCountMapMap `json:"errors"`
 
 	// CollectionErrors is the errors that accumulated while collecting the
 	// status
-	CollectionErrors []error
+	CollectionErrors []error `json:"collection_errors"`
```

Comment:
Since you've added tags, how about skipping null fields with `omitempty` attribute?

Reply:
Only for those fields that are not populated by default, or can show empty values like {} (e.g., list).

Reply:
Right, so no `int` and other mandatory values then.